packages/manager/src/managers/prismicRepository/PrismicRepositoryManager.ts

```diff
@@ -0,0 +1,266 @@
+import * as t from "io-ts";
+import fetch, { Response } from "node-fetch";
+
+import { decode } from "../../lib/decode";
+import { serializeCookies } from "../../lib/serializeCookies";
+
+import { SLICE_MACHINE_USER_AGENT } from "../../constants/SLICE_MACHINE_USER_AGENT";
+import { API_ENDPOINTS } from "../../constants/API_ENDPOINTS";
+
+import { BaseManager } from "../BaseManager";
+
+import {
+	PrismicRepository,
+	PrismicRepositoryRole,
+	PrismicRepositoryUserAgent,
+	PrismicRepositoryUserAgents,
+} from "./types";
+
+const DEFAULT_REPOSITORY_SETTINGS = {
+	plan: "personal",
+	isAnnual: "false",
+	role: "developer",
+};
+
+type PrismicRepositoryManagerCheckExistsArgs = {
+	domain: string;
+};
+
+type PrismicRepositoryManagerCreateArgs = {
+	domain: string;
+	framework: string; // TODO: Type(?)
+};
+
+type PrismicRepositoryManagerDeleteArgs = {
+	domain: string;
+	password: string;
+};
+
+type PrismicRepositoryManagerPushDocumentsArgs = {
+	domain: string;
+	signature: string;
+	documents: Record<string, unknown>; // TODO: Type unknown if possible(?)
+};
+
+export class PrismicRepositoryManager extends BaseManager {
+	// TODO: Add methods for repository-specific actions. E.g. creating a
+	// new repository.
+
+	async readAll(): Promise<PrismicRepository[]> {
+		const url = new URL("./repositories", API_ENDPOINTS.PrismicUser);
+		const res = await this._fetch({ url });
+		const json = await res.json();
```

Comment:
I guess throws are handled in the consumer

Reply:
Since we are talking with a Prismic API, we make some assumptions that the API will return valid JSON.

In general, handling these types of errors at the edge is preferred IMO. That lets us provide a helpful message and display it appropriately (e.g. showing it in a CLI app may be different than in SM UI).

Reply:
ok